homeworks/week6/Minh-Le.ts

```diff
@@ -0,0 +1,94 @@
+//repl: https://repl.it/@MinhD1/minimumSubstring#index.ts
+
+class Main {
+    static test() {
+      console.log(Main.getShortestSubstringOfAllCharacters('xyyzyzyx', 'xyz')); //zyx
+  
+      console.log(Main.getShortestSubstringOfAllCharacters('jzyfgyzxuzyyuuuxz', 'xyyzz')); // yzxuzy
+  
+      console.log(Main.getShortestSubstringOfAllCharacters('a', 'a')); // a
+  
+      console.log(Main.getShortestSubstringOfAllCharacters('fgzyyxdgsh', 'xyyzz')); // ''
+  
+      console.log(Main.getShortestSubstringOfAllCharacters('zfgyyxdgshz', 'xyyzz')); // 'zfgyyxdgshz'
+    }
+  
+    static getShortestSubstringOfAllCharacters(arr: string, str: string): string {
```

Comment:
I forgot to make a note to the problem that arr has unique characters. But your solution works for when it has non-unique char as well.

The solution using sliding window by moving left and right while maintaining a mapCount is an optimal solution.

Good work!